tsdb/points.go

```diff
@@ -357,6 +380,62 @@ func scanTime(buf []byte, i int) (int, []byte, error) {
 	return i, buf[start:i], nil
 }
 
+func isNumeric(b byte) bool {
+	return (b >= '0' && b <= '9') || b == '.'
+}
+
+// scanNumber returns the end position within buf, start at i after
+// scanning over buf for an integer, or float.  It returns an
+// error if a invalid number is scanned.
+func scanNumber(buf []byte, i int) (int, []byte, error) {
+	start := i
+
+	// Is negative number?
+	if i < len(buf) && buf[i] == '-' {
```

Comment:
No.  Just skipping over the `-` if it starts with one.  If we see a `-` again, it'll return an error unless it's a number in scientific notation like `1.0e-3`.